service/src/main/java/bio/terra/workspace/db/StateDao.java

```diff
@@ -0,0 +1,158 @@
+package bio.terra.workspace.db;
+
+import bio.terra.common.exception.ErrorReportException;
+import bio.terra.workspace.common.exception.InternalLogicException;
+import bio.terra.workspace.db.exception.ResourceStateConflictException;
+import bio.terra.workspace.db.model.DbStateful;
+import bio.terra.workspace.service.resource.model.WsmResourceState;
+import javax.annotation.Nullable;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Component;
+
+/** State DAO provides common methods for inspecting and updating state. */
+@Component
+public class StateDao {
+  private static final Logger logger = LoggerFactory.getLogger(StateDao.class);
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  public StateDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    this.jdbcTemplate = jdbcTemplate;
+  }
+
+  /**
+   * Check if the resource is already in the target state with the matching flight id. There are two
+   * cases. In the NOT_EXISTS case, we see if the resource is not there. In all other cases, we see
+   * if the resource is there in the expected state.
+   *
+   * @param dbStateful data fetched from the object row or null
+   * @param state expected state
+   * @param flightId expected flightId
+   * @return true if the resource is in the expected state; false otherwise
+   */
+  public boolean isResourceInState(
+      @Nullable DbStateful dbStateful, WsmResourceState state, String flightId) {
+    if (dbStateful == null) {
+      return (state == WsmResourceState.NOT_EXISTS && flightId == null);
+    }
+
+    return (dbStateful.getState() == state
+        && StringUtils.equals(dbStateful.getFlightId(), flightId));
+  }
+
+  public void updateState(
```

Comment:
I think this needs `@WriteTransaction`

Reply:
I renamed the module `StateDbUtils`. It is always called from other DAOs that should have the proper transaction markings. I also added a comment about that.